compiler/rustc_resolve/src/macros.rs

```diff
@@ -708,15 +736,29 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                 // expanded into a dummy fragment for recovery during expansion.
                 // Now, post-expansion, the resolution may succeed, but we can't change the
                 // past and need to report an error.
-                // However, non-speculative `resolve_path` can successfully return private items
+                // Special cases:
+                // 1. non-speculative `resolve_path` can successfully return private items
                 // even if speculative `resolve_path` returned nothing previously, so we skip this
                 // less informative error if the privacy error is reported elsewhere.
+                // 2. issue #118455, unresolved top level attribute error didn't imported prelude and
+                // already have emitted an error, report builtin macro and attributes error by the way,
+                // so `check_invalid_crate_level_attr` in can ignore them.
+
                 if this.privacy_errors.is_empty() {
-                    this.dcx().emit_err(CannotDetermineMacroResolution {
-                        span,
-                        kind: kind.descr(),
-                        path: Segment::names_to_string(path),
-                    });
+                    if this.is_builtin_macro(res)
+                        || this.builtin_attrs_bindings.values().any(|b| b.res() == res)
```

Comment:
What these two conditions try to achieve?
Why is `CannotDetermineMacroResolution` reporting put under them?
`CannotDetermineMacroResolution` should be reported for anything, not just built-in macros and attributes.

Also, I don't think we ever going to report `CannotDetermineMacroResolution` for built-in *attributes*, they are always reported successfully.